src/main/java/ch/njol/skript/lang/ExpressionList.java

```diff
@@ -78,37 +82,37 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 
 	@Override
 	@Nullable
-	public T getSingle(Event e) {
+	public T getSingle(Event event) {
 		if (!single)
 			throw new UnsupportedOperationException();
 		Expression<? extends T> expression = CollectionUtils.getRandom(expressions);
-		return expression != null ? expression.getSingle(e) : null;
+		return expression != null ? expression.getSingle(event) : null;
 	}
 
-	@SuppressWarnings("unchecked")
 	@Override
-	public T[] getArray(Event e) {
+	@SuppressWarnings("unchecked")
+	public T[] getArray(Event event) {
 		if (and)
-			return getAll(e);
+			return getAll(event);
 		Expression<? extends T> expression = CollectionUtils.getRandom(expressions);
-		return expression != null ? expression.getArray(e) : (T[]) Array.newInstance(returnType, 0);
+		return expression != null ? expression.getArray(event) : (T[]) Array.newInstance(returnType, 0);
 	}
 
-	@SuppressWarnings({"null", "unchecked"})
 	@Override
-	public T[] getAll(Event e) {
-		ArrayList<T> r = new ArrayList<>();
+	@SuppressWarnings("unchecked")
+	public T[] getAll(Event event) {
+		List<T> list = new ArrayList<>();
 		for (Expression<? extends T> expr : expressions)
-			r.addAll(Arrays.asList(expr.getAll(e)));
-		return r.toArray((T[]) Array.newInstance(returnType, r.size()));
+			list.addAll(Arrays.asList(expr.getAll(event)));
+		return list.toArray((T[]) Array.newInstance(returnType, list.size()));
```

Comment:
Might as well use streams. Something like this:
```java
return Arrays.stream(expressions)
		.flatMap(expression -> Arrays.stream(expression.getAll(event)))
		.toArray(...);
```

Reply:
No you cannot do this, incorrect cast.